Amadeus_masterfile.R

```diff
@@ -146,31 +189,36 @@ fun_read_by_country <- function(filename_list, country_name_list, filename_nuts_
     cdata$NUTS_2 <- NA
     cdata$NUTS_1 <- NA
   }
+
+  # 3. Assign deflators by matching country, year, and NACE codes to industry level deflators from EUKLEMS
+  
+  load("DEF_KLEMS_2017ii.Rda")  # reads DataFrame object all_p_ind with columns c("nace2", "def_cd", "ctry", "year", "p_ind_va", "p_ind_go", "p_ind_cp")
+  all_p_ind <- all_p_ind[all_p_ind$ctry==country_abbrv,]    # select country in deflator data frame
+  all_p_ind$ctry <- NULL                                    # remove unused variables
+  all_p_ind$def_cd <- NULL
+  all_p_ind$nace2 <- as.numeric(all_p_ind$nace2)            # change NACE code to numeric to match firm data file structure
```

Comment:
all_p_ind$def_cd <- all_p_ind$def_cd /100 and you're done

Reply:
Also be careful when changing codes to numeric because some start with 0, which is lost in the conversion (but I don't know how the codes are formatted in your data).

Reply:
I forgot to address this in my earlier reply: As far as I see, the NACE codes are numeric in the firm data file. 

    > typeof(cdata$NACE_PRIM_CODE)
    [1] "double"
    > table(cdata$NACE_PRIM_CODE)
    
     113  125  128  130  141  142  150  170  210  240  311  610  729  812  891  899 
       1    2    3    1    1    2    1    2   12    1  344    2    1    1    1   69 
    1000 1011 1020 1032 1041 1071 1082 1089 1101 1104 1105 1107 1399 1413 1419 1420 
       3    2   19    2    1   15    1    6    2    7    4    9   11    8   18    1 
    1520 1610 1623 1629 1712 1722 2010 2012 2016 2030 2041 2222 2332 2349 2361 2362 
      25    6    1   12    6    4    1    1    2    1    2    4    2    1    2   14 
    (...)
    8621 8622 8623 9200 9310 9312 9329 9529 9602 9603 9609 
       4    2    3   16    4    4   20    1    1    1   18 

Note the missing leading zeros for low NACE codes. I think this is because `read.table::fread` does recognize columns with only numeric entries and convert them to numeric types automatically. It may be debatable whether this is the right way to read the data. As long as we only need to identify and compare NACE codes, this this is not important. Also, I am not sure how to coerce `read.table::fread` to do characters for some columns but not for others.

Anyway, with the current way to read the firm data files (with `read.table::fread`), I think this is the right way to compare the NACE codes.

Reply:
cool, happy with that!